source/lambda/helper/python/kendraHelper.py

```diff
@@ -0,0 +1,162 @@
+    #   documentId:             the document id generated by DUS
+    #
+    def deindexDocument(self,
+                        region,
+                        kendraIndexId,
+                        documentId):
+        
+        kendraclient = client = boto3.client('kendra', region_name=region)
+
+        response = client.batch_delete_document(IndexId=kendraIndexId,
+                                                DocumentIdList=[documentId])
+                                                
+        return
+
+    #
+    #   This function seaches Kendra using a natural language query string and a
+    #   user membership tag (healthprovider, scientist, generalpublic)
+    #
+    #   region:                 region of Kendra index
+    #   kendraIndexId:          Kendra index id
+    #   requestBody:            POST body of json search, see example below.
+    #
+    #   { "query":"my keywords",
+    #     "tag":"scientist",
+    #     "pageNumber":1,       pagination is done by providing the page number needed in each request
+    #     "pageSize":100        each page may have a maximum of 100 results
+    #   }
+    #
+    def search(self,
+               region,
+               kendraIndexId,
+               requestBody):
+
+        search = json.loads(requestBody)
+
+        client = client = boto3.client('kendra', region_name=region)
+
+        if 'tag' in search and search['tag'] != None:
+            response = client.query(
+                QueryText=search['query'],
+                IndexId=kendraIndexId,
+                AttributeFilter={
+                    "OrAllFilters": [
+                        {
+                            "EqualsTo": {
+                            "Key": "_group_ids",
+                            "Value": {
+                                "StringListValue": [search['tag']]
+                                }
+                            }
+                        }
+                    ]
+                },
+                PageNumber=search['pageNumber'],
+                PageSize=search['pageSize']
+            )
+
+        else:
+            response = client.query(
+                QueryText=search['query'],
+                IndexId=kendraIndexId,
+                PageNumber=search['pageNumber'],
+                PageSize=search['pageSize']
+            )
+
+        return response
+
+    #
+    #   This function tells Kendra that a specific search result from a previous
+    #   results set is relevant or not. Kendra will use this hint in subsequent
+    #   searches.
+    #
+    #   region:                 region of Kendra index
+    #   kendraIndexId:          Kendra index id
+    #   requestBody:            POST json body of feedback,  see example below
+    #
+    #   {  "queryId":"4c97e09a-5a97-4d3a-beb6-9362fb90fa16",
+    #      "resultId":"4c97e09a-5a97-4d3a-beb6-9362fb90fa16-df5306d5-085d-4c51-8eaf-4add4848643b",
+    #      "relevance":true
+    #   }
+    #
+    def feedback(self,
```

Comment:
I would suggest using verbs for function rather than nouns.
While feedback may be obvious to those using Kendra, a person new to Kendra might not understand the action of this function (For eg is this function to provideFeedback / retreiveFeedback?)